tests/test_math_interpolator.hh

```diff
@@ -0,0 +1,158 @@
+  static double
+  compute_intp_derivative_error(std::shared_ptr<ScalarInterpolator> & intp,
+                                std::function<double(double)> derivative_func,
+                                const math::Vector_Ref & ref_points) {
+    math::Vector_t intp_vals = math::Vector_t::Zero(ref_points.size());
+    math::Vector_t intp_refs = math::Vector_t::Zero(ref_points.size());
+    for (int i{0}; i < ref_points.size() - 1; i++) {
+      intp_vals(i) =
+          convert_to_scalar(intp->interpolate_derivative(ref_points(i)));
+      intp_refs(i) = derivative_func(ref_points(i));
+    }
+    return ErrorMethod::compute_global_error(Vector_Ref(intp_vals),
+                                             Vector_Ref(intp_refs));
+  }
+
+  template <class ErrorMethod, class ScalarInterpolator>
+  static double
+  compute_intp_finite_diff_error(std::shared_ptr<ScalarInterpolator> & intp,
+                                 std::function<double(double)> derivative_func,
+                                 const math::Vector_Ref & ref_points) {
+    Vector_t intp_refs = Vector_t::Zero(ref_points.size() - 1);
+    Vector_t finite_diff_derivative = Vector_t::Zero(ref_points.size() - 1);
+    for (int i{0}; i < ref_points.size() - 2; i++) {
+      intp_refs(i) = derivative_func(ref_points(i));
+      finite_diff_derivative(i) =
+          (convert_to_scalar(intp->interpolate(ref_points(i + 1))) -
+           convert_to_scalar(intp->interpolate(ref_points(i)))) /
+          (ref_points(i + 1) - ref_points(i));
+    }
+    return ErrorMethod::compute_global_error(Vector_Ref(finite_diff_derivative),
+                                             Vector_Ref(intp_refs));
+  }
+
+  template <class ErrorMethod>
+  static double
+  compute_finite_diff_error(std::function<double(double)> func,
+                            std::function<double(double)> derivative_func,
+                            const math::Vector_Ref & ref_points) {
+    Vector_t derivative_refs = Vector_t::Zero(ref_points.size() - 1);
+    Vector_t finite_diff_derivative = Vector_t::Zero(ref_points.size() - 1);
+    for (int i{0}; i < ref_points.size() - 2; i++) {
+      derivative_refs(i) = derivative_func(ref_points(i));
+      finite_diff_derivative(i) =
+          (func(ref_points(i + 1)) - func(ref_points(i))) /
+          (ref_points(i + 1) - ref_points(i));
+    }
+    return ErrorMethod::compute_global_error(Vector_Ref(finite_diff_derivative),
+                                             Vector_Ref(derivative_refs));
+  }
+
+  template <class Interpolator>
+  struct InterpolatorFixture {
+    using RadialIntegral_t =
+        internal::RadialContribution<internal::RadialBasisType::GTO>;
+    InterpolatorFixture<Interpolator>() {}
+    typedef Interpolator Interpolator_t;
+
+    double x1{0};
+    double x2{1};
+    double error_bound{1e-5};
+    int nb_ref_points{100};
+
+    std::map<std::string, std::function<double(double)>> functions{
```

Comment:
Could this constructs also be a json? Ah, no, I see you have the functions in here for tests.
Any reasons why exactly those functions were chosen? Maybe elaborate in 1-2 sentences.